### DISCUSSION TITLE
[Multiplayer] Campaign Player Characters Should Retain Skill Levels on Respawn

### DISCUSSION THREAD

**Discussion by DrewTuber** [FakeFishGames/Barotrauma#4795] [Ideas and suggestions]
Over long campaigns players will spend a lot of time getting XP from repairing and firing weapons at the many denizens of Europa. Leveling up is slow, yet satisfying. What ISN'T satisfying is losing all that XP and having a stat reroll every time a player dies. This strongly incentivizes restarting missions when a single crew mate is lost instead of trying to salvage the situation and pressing on.

**Comment by Kulik242** [comment]
On the other hand, you will cling to your life very tenaciously if you have that much to loose, bringing things like cowardice and valor quite nicely into the game and give more meaning to in-game selfsacrifice.

Perhaps, there should be a difference, whether the body is recovered and brought to an outpost. Player dying and not recovering the body would mean a complete restart for him, recovering the body would result in a big XP hit but not a complete restart.  The XP hit may be proportional to the duration the player stays in the "decased" status, thus the crew would have to make a call to abandoning the mission and hurry to an outpost or sacrifice more of that character skill.
I actually think this would introduce a new interesting aspect into the game.

**Comment by NilanthAnimosus** [comment]
If you die, collect your items and get keep your skills - there is effectively no penalty to dying really, short of being eaten causing the item loss quite often your capable of retrieving a body and your items.

I feel the skill loss is sort of needed, at most this should only ever be a none-default option a campaign can set. I do feel however for player characters in an MP campaign the starter skills should be slightly less randomized however, I never liked starting with some 50 medical one doctor then like, 43 another.

I think once a person is dead, well their dead.

If one were to alleviate the penalty through some method, such as kulik242's concept of recovering the body being the decision between full skill loss or partial (And partial being effected by time spent dead) - Instead I would suggest given respawning could be a thing later or not a way to mark/set a deceased player to have (at monetary cost) a portion of their skills preserved if they do not change class.

it could be implemented as a game/crew option that can be changed anytime or set per-player as a sort of insurance you pay on revival next round (or shuttle respawn if a thing) to keep a set amount of your skill - and definitely not all of it, but not losing it because you were eaten by something or forcing your time to be cut short / interfering with other ideas like respawn shuttles if ever added to campaigns.

Finally, any skill loss should definately not go below the jobs default values, otherwise dying could result in the lowest possible job skill values and be worse than simply losing your skills outright. care should be taken on percentage based skill losses.

**Comment by Ek-Videogames** [comment]
(This should be a server setting probably. I do have some sympathy for _'those people'_ that try to main assistant as their only class and then just die to a mudraptor 10 minutes later. )

"This strongly incentivizes restarting missions when a single crew mate is lost instead of trying to salvage the situation and pressing on."
 (Or using modded executable to cheat character skills, and if people get killed by BS bugs or desyncs or disconnect death)

**Comment by RadCatish** [comment]
What about if instead of retaining skill levels, the levels of the respawn character scale with the difficulty of the current stage of the campaign? So rather than spawning with a very green character on a very hard leg, they have slightly higher starting levels than those at the beginning of the campaign. This could be justified in-world by these "new recruits" coming from frontier outposts which are deeper and more dangerous than those the crew came from at the beginning of the campaign
